tests/lib/rules/no-shadow-route-definition.js

```diff
@@ -0,0 +1,198 @@
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require('../../../lib/rules/no-shadow-route-definition');
+const RuleTester = require('eslint').RuleTester;
+
+const { ERROR_MESSAGE } = rule;
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({
+  parser: require.resolve('babel-eslint'),
+  parserOptions: {
+    ecmaVersion: 6,
+    sourceType: 'module',
+  },
+});
+
+ruleTester.run('no-shadow-route-definition', rule, {
+  valid: [
+    'this.route("blog");',
+    'this.route("blog", function() {});',
+    'this.route("blog", { ...foo });',
+    'this.route("blog", { path: undefined });',
+    'this.route("blog", { path: "" });',
+    'this.route("blog", { path: "/" });',
+    'this.route("blog", { path: "blog-posts" });',
+    'this.route("blog", { path: "blog-posts" }, function() {});',
+    'this.route("blog", { path: "/blog-posts" });',
+    'this.route("blog", { path: "blog-posts", otherOption: true });',
+    `this.route("blog", { path: "/" }, function() {
+      this.route("sub-blog", { path: "/" });
+    });`,
+    `this.route("blog", function() {
+      this.route("post");
+    });`,
+    `this.route("blog", { path: "/" }, function() {
+      this.route("post");
+    });`,
+    `this.route("blog", { path: "/" }, function() {
+      this.route("post", { path: "hero" });
+    });`,
+    `this.route("blog", function() {
+      this.route("post", { path: "hero" });
+    });`,
+    `this.route("blog", { path: "/blog/post" }, function() {
+      this.route("post", { path: "/" });
+    });
+    this.route("post")`,
+    `this.route("blog", function() {
+      this.route("post", { path: "/" });
+    });
+    this.route("post")`,
+
+    // With dynamic segment:
+    'this.route("blog", { path: ":blog" });',
```

Comment:
Actually this is incorrect for this rule, as both path definitions would equal `blog/<value>` 🤔 . I think I need to work a bit more on this.